src/twisted/__main__.py

```diff
@@ -0,0 +1,17 @@
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+Make the twisted module executable with the default behaviour of
+running twist.
+"""
+
+from __future__ import division, absolute_import
+
+import sys
+from pkg_resources import load_entry_point
+
+if __name__ == '__main__':
```

Comment:
I don't think there is any protection against importing `__main__`: at least, I am capable of doing it for tools like `pip`. So I think we still want the if branch. 

What we should probably do is test that no output is produced when importing the `__main__` module directly, in this case by delaying the import until a test case and then replicating [this test logic](https://github.com/twisted/twisted/blob/trunk/src/twisted/trial/test/test_warning.py#L161-L165) which mocks out stdout.